metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -457,9 +466,13 @@ private Throwable handleEventException(String name,
         long endProcessingTime = time.nanoseconds();
         long deltaNs = endProcessingTime - startProcessingTimeNs.getAsLong();
         long deltaUs = MICROSECONDS.convert(deltaNs, NANOSECONDS);
-        if (exception instanceof ApiException) {
+        if ((exception instanceof ApiException) ||
+                (exception instanceof BoundedListTooLongException)) {
             log.info("{}: failed with {} in {} us. Reason: {}", name,
                 exception.getClass().getSimpleName(), deltaUs, exception.getMessage());
+            if (exception instanceof BoundedListTooLongException) {
+                exception = new UnknownServerException(exception.getMessage());
```

Comment:
This is pretty uninformative. Can we use `PolicyViolationException` with a string indicating the specifics? Not perfect, but the ability to include a custom string is super helpful.